### PR TITLE
Update isclose based on decisions made

### LINKED ISSUE #19025: Math module - adjustments to isclose
`isclose` should be camelCased.  Also, we should document that the argument `rtol` is for relative tolerance and the argument `atol` is for absolute tolerance.

### PR DESCRIPTION
**Pull request by lydia-duncan**
This handles the decisions made related to this function:
- Renames it to comply with our naming guidelines (to `isClose`)
- Renames its `rtol` argument to `relTol`
- Renames its `atol` argument to `absTol`
- Documents the `relTol` and `absTol` arguments
- Limits the types accepted for `x` and `y` to `real`, `imag` and `complex`

Resolves #19025 

Deprecates the old version of the function.

Adds a new test locking in the deprecation message, and updates
the original test of the feature to use the new names and limitations.
Also updates a couple of tests we wrote that were relying on the
old function and its behavior.

Passed a full paratest with futures

### CODE REVIEW

**Review by jabraham17 — APPROVED** [review]
Looks good to me! Just left a few comments

File: modules/standard/AutoMath.chpl [hunk 1]
```diff
@@ -2515,7 +2515,28 @@ module AutoMath {
...
+    if boundsChecking && (relTol < 0) then
+      HaltWrappers.boundsCheckHalt("Input value for relTol must be positive");
+    if boundsChecking && (absTol < 0) then
+      HaltWrappers.boundsCheckHalt("Input value for absTol must be positive");
+    var diff: real = abs(x-y);
+    return ( (diff<=abs(relTol*y)) || (diff<=abs(relTol*x)) || (diff<=absTol) );
+  }
+
   /* Returns true if `x` and `y` are approximately equal, else returns false. */
+  @deprecated("isclose with 'rtol' and 'atol' arguments is now deprecated, please use :proc:`isClose` with 'relTol' and 'absTol' arguments instead")
```
[the remark this answers is not in the document]
Comment (lydia-duncan) [reply modules/standard/AutoMath.chpl:2539]: Not as easily as the previous PR, due to changes made to the error messages to account for the new names

[the remark this answers is not in the document]
Comment (lydia-duncan) [reply modules/standard/AutoMath.chpl:2539]: The new version also supports more limited types, so calling it with other types wouldn't behave the same as it did prior to the deprecation

**Comment by lydia-duncan** [comment]
I've added some tests in addition to the change mentioned, please take a look :)

**Comment by jabraham17** [comment]
Looks good!

**Pull request merged**